resources/components/base/Suggestion.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+	<div
+		class="mw-suggestion"
+		tabindex="0"
+		v-bind:class="classObject"
+		v-on:click="onClick"
+		v-on:keydown.enter="onClick"
+	>
+		<label class="mw-suggestion__label">
+			<slot />
+		</label>
+		<icon
+			v-if="confirmed"
+			icon="check"
+			v-bind:title="iconText"
+			v-bind:label="iconText"
+		/>
+	</div>
+</template>
+
+<script>
+var Icon = require( './Icon.vue' );
+
+module.exports = {
+	name: 'Suggestion',
+
+	components: {
+		icon: Icon
+	},
+
+	props: {
+		text: {
+			type: String,
+			required: true
+		},
+		confirmed: {
+			type: Boolean,
+			default: false
+		}
+	},
+
+	data: function () {
+		return {
+			iconText: this.$i18n( 'machinevision-suggestion-confirm-undo-title', this.text ).parse(),
+			toggled: false
+		};
+	},
+
+	computed: {
+		classObject() {
+			return {
+				'mw-suggestion--confirmed': this.confirmed,
+				'mw-suggestion--toggled': this.toggled
+			};
+		}
+	},
+
+	methods: {
+		onClick: function () {
+			// Adding the toggled class only after a suggestion has been toggled
+			// once, that is, the first time it is selected, will allow us to
+			// animate the process of un-selecting a suggestion (sliding it back
+			// to center) without firing off that animation when the component
+			// first mounts.
+			this.toggled = true;
```

Comment:
So just to be clear, "toggled" is not intended to replace the suggestion's "confirmed" state which is managed by the parent component? At first I thought it might be redundant but it sounds like it's only used for animation?

In a normal situation (animations aside), we'd want these widgets to be as "dumb" as possible – only emitting events upwards to that the parent and not changing their state themselves. This is because the ImageCard widget needs to know the confirmed state of all the suggestions in order to make the correct API request on publish.

Reply:
Correct, it's just for animation. Suggestions are animated on confirm (text moved to the left to make room for the checkmark) and on un-confirm (text moved back to center). If we just add animation to these two states, the un-confirm animation will run on page load. To get around this I'm adding a class after the suggestion has been clicked once, because from then on we want to animate the suggestion in its un-confirmed state.

Reply:
Is it possible to avoid this problem by replacing the animations with transitions? It seems to me like something like the below should work (although admittedly I haven't tested this):
```less
.mw-suggestion {
    .mw-icon {
        .transition( opacity 0.2s );
    }

    &--toggled {
        .mw-icon {
            opacity: 0;
        }
    }

    &__label {
        .transition-transform( 0.2s );
    }

    &--confirmed {
        .mw-suggestion__label {
            transform: translateX( -0.5em );
        }
    }
```

Reply:
Yeah, this is a much better solution. I think I got `animation`-happy because I was using it for other elements when they appear in the DOM (e.g. toasts), but `transition` makes way more sense when there's a class name change to trigger them. Thanks!

Reply:
Cool – I think that polluting the state with something we only need for presentation should be avoided as much as possible.